src/jvmMain/kotlin/io/ktor/io/JvmBufferPool.kt

```diff
@@ -1,11 +1,13 @@
 package io.ktor.io
 
+import java.nio.ByteBuffer
 
 public class JvmBufferPool(
     capacity: Int = DEFAULT_POOL_CAPACITY,
+    public val byteBufferPool: ObjectPool<ByteBuffer> = ByteBufferPool.Default
```

Comment:
minor: in byteArrayBufferPool, underlying buffer goes before capacity
minor: Empty pool can be implemented in the same way as in byteArrayBufferPool